kinesis/kinesis.go

```diff
@@ -142,6 +151,8 @@ func NewOutputPlugin(region, stream, dataKeys, partitionKey, roleARN, kinesisEnd
 		random:                random,
 		Concurrency:           concurrency,
 		concurrencyRetryLimit: retryLimit,
+		isAggregate:           isAggregate,
+		aggregator:            aggregator,
```

Comment:
@zackwine Is aggregation compatible with your concurrency feature? Does storing the aggregator on the plugin struct mean it can't be re-used between goroutines?

Reply:
I think its fine if it isn't, but if those options are incompatible, the plugin shouldn't allow users to config both of them

Reply:
Never mind, I realized all of this takes place in the unpackRecords function before any goroutines are created

Reply:
Yes, the `FlushAggregatedRecords` method was added to ensure aggregation was compatible with concurrency.